models/repo.go

```diff
@@ -2363,3 +2401,28 @@ func (repo *Repository) CreateNewBranch(doer *User, oldBranchName, branchName st
 
 	return nil
 }
+
+// CreateNewBranchFromCommit creates a new repository branch
+func (repo *Repository) CreateNewBranchFromCommit(doer *User, commit, branchName string) (err error) {
```

Comment:
Please put this function in `models/repo_branch.go` instead. Let's not add more to the God File unless we have to 😄